bluemira/magnets/materials.py

```diff
@@ -0,0 +1,463 @@
+# SPDX-FileCopyrightText: 2021-present M. Coleman, J. Cook, F. Franza
+# SPDX-FileCopyrightText: 2021-present I.A. Maione, S. McIntosh
+# SPDX-FileCopyrightText: 2021-present J. Morris, D. Short
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""
+Material database used for magnets
+"""
+
+import math
+from abc import ABC, abstractmethod
+
+
+class ABCMaterial(ABC):
+    """Abstract Material class"""
+
+    @abstractmethod
+    def density(self, **kwargs):
+        """Density"""
+        pass
+
+    @abstractmethod
+    def E(self, **kwargs):  # noqa: N802
+        """Young's module"""
+        pass
+
+    @abstractmethod
+    def erho(self, **kwargs):
+        """Electrical Resistivity"""
+        pass
+
+    @abstractmethod
+    def cp_v(self, **kwargs):
+        """Specific heat (constant volume)"""
+        pass
+
+
+class Material(ABCMaterial):
+    """Generic Material"""
+
+    def density(self, **kwargs):  # noqa: ARG002
+        """Density"""
+        return 0
+
+    def E(self, **kwargs):  # noqa: N802, ARG002
+        """Young's module"""
+        return 1e-6
+
+    def erho(self, **kwargs):  # noqa: ARG002
+        """Electrical Resistivity"""
+        return 1e6
+
+    def cp_v(self, **kwargs):  # noqa: ARG002
```

Comment:
Potential for some confusion here. Is this the volumetric heat capacity or the specific heat capacity? So extensive of intensive? I think in general (and something that I regret a little with the present materials implementation in bluemira) that it would be nice to have full_word_names for these properties, even if sometimes they will look horrid. Saves on potential confusion of this nature. I also find the signature (temperature: float, field: Union[None, float], strain: Union[None, float]) perhaps a little easier to read.